src/appleseed/renderer/modeling/bsdf/hairbsdf.cpp

```diff
@@ -484,9 +498,9 @@ namespace
             float gammaO = std::asin(clamp(values->m_precomputed.m_h, -1.0f, 1.0f));
 
             // Compute Attenuation for the single path through the hair cylinder
-            Spectrum T = Exp(values->m_precomputed.m_sigma_a, 2 * cosGammaT / cosThetaT);
+            Spectrum T = Exp(values->m_precomputed.m_sigma_a, 2.f * cosGammaT / cosThetaT);
 
-            Vector2f u[2] = {demuxFloat(s[0]),demuxFloat(s[1])};
+            Vector2f u[2] = {demuxFloat(s1[0]),demuxFloat(s1[1])};
```

Comment:
You can ask the sampling context 4 samples. In fact some BSDFs (Glass) used to do it before.
The result is probably going to be a bit better and you can get rid of demuxFloat.